scripts/travis-presubmit.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# Checks for lint errors, spelling, licensing and so on.
+set -eu
+
+check_deps() {
+  local failed=0
+  check_cmd golint github.com/golang/lint/golint || failed=10
+  check_cmd errcheck github.com/kisielk/errcheck || failed=11
+  check_cmd gocyclo github.com/fzipp/gocyclo || failed=12
+  check_cmd ineffassign github.com/gordonklaus/ineffassign || failed=13
+  check_cmd misspell github.com/client9/misspell/cmd/misspell || failed=14
+  return $failed
+}
+
+check_cmd() {
+  local cmd="$1"
+  local repo="$2"
+  if ! type -p "${cmd}" > /dev/null; then
+    echo "${cmd} not found, try to 'go get -u ${repo}'"
+    return 1
+  fi
+}
+
+main() {
+  check_deps
+
+  local fail=0
+  local go_srcs="$(find . -name '*.go' | \
+    grep -v mock_ | \
+    grep -v .pb.go | \
+    grep -v .pb.gw.go | \
+    grep -v _string.go | \
+    grep -v vendor/ | \
+    tr '\n' ' ')"
+  local proto_srcs="$(find . -name '*.proto' | \
+    grep -v vendor/ | \
+    tr '\n' ' ')"
+
+  printf "running gofmt...\n"
+  find . -iregex '.*.go' ! -path "./vendor/*" -exec gofmt -s -w {} \;
+  status=$(git status --porcelain)
+  if [[ -n ${status} ]]; then
+    printf "gofmt changed the following files:\n${status}\n"
+    git status
+    git diff
+    fail=1
+  fi
+
+  printf "running golint...\n"
+  status=$(find . -iregex '.[^.]*.go' ! -name 'gen.go' ! -path "./vendor/*" -exec golint {} \;)
+  if [[ -n ${status} ]]; then
+    printf "golint found the following issues:\n${status}\n"
+    fail=1
```

Comment:
I've just learned that `golint` doesn't return a non-zero status if it found any issues. This means neither the code above nor the changed line in c3d66f8 would fail or produce any output. If you want at least to see `golint`'s output you can add a `set_exit_status` flag to its call.